youtube_dl/extractor/sverigesradio.py

```diff
@@ -0,0 +1,130 @@
+                url = audiourl['audioUrl']
+                if url == "" or url in audiourls:
+                    continue
+
+                format = {'vcodec': 'none', 'url': url}
+                # add codec and bitrate if known coding format
+                codingformat = audiourl.get('codingFormat')
+                if codingformat:
+                    format.update(self._CODING_FORMATS.get(codingformat, {}))
+
+                audiourls[url] = format
+
+        return audiourls.values()
+
+    def _extract_audio(self, first_type, secondary_type, url):
+        audio_id = self._match_id(url)
+        query = {'id': audio_id, 'type': first_type}
+
+        metadata = self._download_json(self._BASE_URL + '/audiometadata', audio_id, query=query)
+        item = metadata['items'][0]
+
+        formats = self._extract_formats(query, audio_id, first_type, secondary_type)
+
+        return {
+            'id': audio_id,
+            'title': item.get('subtitle'),
+            'formats': formats,
+            'series': item.get('title'),
+            'duration': int_or_none(item.get('duration')),
+            'thumbnail': item.get('displayimageurl'),
+            'description': item.get('description'),
+        }
+
+
+class SverigesRadioPublicationIE(SverigesRadioBaseIE):
+    _VALID_URL = r'https?://(?:www\.)?sverigesradio\.se/sida/(?:artikel|gruppsida)\.aspx\?.*artikel=(?P<id>[0-9]+)'
+    _TESTS = [{
+        'url': 'https://sverigesradio.se/sida/artikel.aspx?programid=83&artikel=7038546',
+        'md5': '6a4917e1923fccb080e5a206a5afa542',
+        'info_dict': {
+            'id': '7038546',
+            'ext': 'm4a',
+            'duration': 132,
+            'series': 'Nyheter (Ekot)',
+            'title': 'Esa Teittinen: Sanningen har inte kommit fram',
+            'description': 'md5:daf7ce66a8f0a53d5465a5984d3839df',
+            'thumbnail': 're:^https://static-cdn.sr.se/sida/images/',
+        },
+    }, {
+        'url': 'https://sverigesradio.se/sida/gruppsida.aspx?programid=3304&grupp=6247&artikel=7146887',
+        'only_matching': True,
+    }]
+
+    def _real_extract(self, url):
+        return self._extract_audio('publication', None, url)
+
+
+class SverigesRadioEpisodeIE(SverigesRadioBaseIE):
+    _VALID_URL = r'https?://(?:www\.)?sverigesradio\.se/(?:sida/)?avsnitt/(?P<id>[0-9]+)'
+    _TESTS = [{
+        'url': 'https://sverigesradio.se/avsnitt/1140922?programid=1300',
+        'md5': '20dc4d8db24228f846be390b0c59a07c',
+        'info_dict': {
+            'id': '1140922',
+            'ext': 'mp3',
+            'duration': 3307,
+            'series': 'Konflikt',
+            'title': 'Metoo och valen',
+            'description': 'md5:fcb5c1f667f00badcc702b196f10a27e',
+            'thumbnail': 're:^https://static-cdn.sr.se/sida/images/'
+        }
+    }, {
+        # has type secondary
+        'url': 'https://sverigesradio.se/avsnitt/1257942',
```

Comment:
the duration for `episode` file and `secondary` file is different, if the content of the files is different then a playlist should be used.

Reply:
It seems like one of them is cut from the live broadcast so they differ a bit at the beginning. Not sure if that is ok?

Reply:
it's either one of two:
they are identical -> keep them as they are(formats of the same entry).
they are different -> separate them into a playlist with two entries.

Reply:
Would that mean use `self.playlist_result` (to have `'_type': 'playlist'`) and have multiple entries? but would that make an episode or publication url potentially download a version of each entry when the user probably expected just one file.

Maybe a third option is to skip secondary?

Reply:
unless there is another example that have a `secondary` version that is significantly diffrent version from the `episode` one, i think it can be dropped for now.

Reply:
It seems to quite common. Most of the episodes in the "grunden till allt" series have differences. One even by 30 seconds, there is additional talking in the beginning and the end from the live broadcast.

https://sverigesradio.se/avsnitt/1240091 (~8 sec diff)
https://sverigesradio.se/avsnitt/1240056 (~30 sec diff)

Removing it